### PR TITLE
fix(s3): Extract bucket creation and improved key handling

### LINKED ISSUE #6: Enhancements for Optional Bucket Creation and Improved Key Handling
### Enhancements and Clarifications Needed Based on Latest Pull Request Review by @leosmerling-hopeit:

1. **Bucket Creation as a Method and Optional for Initialization:**
   - Make bucket creation a method within `ObjectStorage`. Note this is not recommended to use in production environments
   - Remove the create_buacket option.

2. **Prefix Handling:**
   - Ensure the prefix always ends with a `/`.
   - Simplify key construction by removing trailing `/` from both `partition_key` and `prefix` before concatenation.
   - Verify that `partition_key` and `prefix` are not already stripped of trailing `/`.

3. **File Name Construction:**
   - Streamline file name construction considering the presence of `partition_key` and `prefix`.

4. **Wildcard and Partition Date Format:**
   - Make consistent default to `False` on `recursive`.
   - Clarify this behavior in the documentation.

### PR DESCRIPTION
**Pull request by pcanto-hopeit**
resolves #6

### CODE REVIEW

File: plugins/aws/s3/src/hopeit/aws/s3/object_storage.py [hunk 1]
```diff
@@ -436,6 +390,46 @@ def partition_key(self, path: str) -> str:
             partition_key = path.rsplit("/", 1)[0]
         return partition_key
 
+    async def create_bucket(self, check_if_exists: bool = True) -> bool:
```
Comment (leosmerling-hopeit) [inline plugins/aws/s3/src/hopeit/aws/s3/object_storage.py]: implement using signature from os.makedirs https://docs.python.org/3/library/os.html#os.makedirs

File: plugins/aws/s3/src/hopeit/aws/s3/object_storage.py [hunk 2]
```diff
@@ -128,12 +122,12 @@ def __init__(
         :param prefix, Optional[str]: Prefix to be used for every element (object or file) stored in the S3 bucket.
         :param partition_dateformat, Optional[str]: Optional format string for partitioning
             dates in the S3 bucket.
-        :param create_bucket, bool: Whether to create the S3 bucket if it doesn't exist
         """
         self.bucket: str = bucket
-        self.prefix: Optional[str] = prefix
+        self.prefix: Optional[str] = (
+            (prefix.rstrip("/") + "/") if prefix and prefix != "/" else None
```
Comment (leosmerling-hopeit) [inline plugins/aws/s3/src/hopeit/aws/s3/object_storage.py]: no need to check prefix != '/', if the backed does not support it it should fail.

File: plugins/aws/s3/src/hopeit/aws/s3/object_storage.py [hunk 3]
```diff
@@ -450,13 +444,12 @@ async def _aioglob(
...
-
             prefix = self.prefix or ""
 
             if wildcard:
-                dir_path = os.path.dirname(wildcard)
-                if dir_path:
-                    prefix = os.path.join(prefix, dir_path) + "/"
+                dir_path = Path(wildcard).parent
+                if dir_path != Path("."):
+                    prefix = f"{prefix}{dir_path}/"
```
Comment (leosmerling-hopeit) [inline plugins/aws/s3/src/hopeit/aws/s3/object_storage.py]: prefix += f"{dir_path}/"

File: plugins/aws/s3/src/hopeit/aws/s3/object_storage.py (hunk 1 above)
Comment (pcanto-hopeit) [reply plugins/aws/s3/src/hopeit/aws/s3/object_storage.py]: address in the last commit!

**Comment by pcanto-hopeit** [comment]
Hi @leosmerling-hopeit,  if everything looks good, feel free to release it.

File: apps/examples/aws-example/src/aws_example/s3/init.py
Comment (leosmerling-hopeit) [inline apps/examples/aws-example/src/aws_example/s3/init.py:1]: This event could/should have a test using execute_event approach like the other events.

**Review by leosmerling-hopeit — APPROVED** [review]

File: apps/examples/aws-example/src/aws_example/s3/init.py
Comment (pcanto-hopeit) [reply apps/examples/aws-example/src/aws_example/s3/init.py:1]: I addressed this by following the `execute_event` approach and also added the missing aws-example `s3.init` SETUP event test.

**Pull request merged**